service_configuration_lib/utils.py

```diff
@@ -28,3 +35,52 @@ def load_spark_srv_conf(preset_values=None) -> Tuple[Mapping, Mapping, Mapping,
     except Exception as e:
         log.warning(f'Failed to load {DEFAULT_SPARK_RUN_CONFIG}: {e}')
         raise e
+
+
+def ephemeral_port_reserve_range(preffered_port_start: int, preferred_port_end: int, ip='127.0.0.1') -> int:
```

Comment:
it'd be nice to upstream this into the ephemeral_port_reserve repo, but we can wait to see how this works for y'all's usecase first if you'd like :)

Reply:
Sure! I added comments explaining the purpose of adding this function and linked to the upstream repository for reference. Let's leave it here for now and test it out first